### DISCUSSION TITLE
How to use HashToG1 with frontend.Variable

### DISCUSSION THREAD

**Discussion by HamzaZF** [Consensys/gnark#1205] [General]
Hello,

I'm a beginner with gnark and I am wondering how I can use hashToG1 with a frontend.Variable type. This functions seems to take bytes as input.

Thank you in advance.

Hamza

**Comment by ivokub** [comment]
Can you post an example circuit what you are trying to do?

NB! It seems to be a question about gnark usage, I'm converting the issue to discussion.

**Reply by HamzaZF** [reply]
Thank you for your reply.

Specifically, I need to verify BLS signatures, but in my case, the message (denoted as cm) is a variable of type frontend.Variable. To verify e(sig, g2)==e(H(cm), pk), I first need to perform a hash to curve on cm. However, the function provided by the library (https://pkg.go.dev/github.com/consensys/gnark-crypto@v0.12.1/ecc/bls12-377#HashToG1) takes bytes as a parameter and not a frontend.Variable type.

Here is link to the circuit in the playground : http://play.gnark.io/?id=vcwbjvubsg

**Reply by ivokub** [reply]
We currently don't have in-circuit method for computing hash to curve method for in-circuit implementation. There is an [open PR adding the support for BLS12-381](https://github.com/Consensys/gnark/pull/1040), but it is not generalized for all curves. You can have a look at the PR and try to adapt to BLS12-377.

We have planned to review #1040 and generalize for all supported curves, but it is quite significant work and haven't gotten to it yet.
